app/team_test.go

```diff
@@ -269,6 +270,68 @@ func TestAddUserToTeamByToken(t *testing.T) {
 		assert.NotNil(t, err)
 	})
 
+	t.Run("invalid add a guest user with a non-granted email domain", func(t *testing.T) {
+		restrictedDomain := *th.App.Config().GuestAccountsSettings.RestrictCreationToDomains
+		defer func() {
+			th.App.UpdateConfig(func(cfg *model.Config) { cfg.GuestAccountsSettings.RestrictCreationToDomains = &restrictedDomain })
+		}()
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.GuestAccountsSettings.RestrictCreationToDomains = "restricted.com" })
+		token := model.NewToken(
+			TOKEN_TYPE_GUEST_INVITATION,
+			model.MapToJson(map[string]string{"teamId": th.BasicTeam.Id, "channels": th.BasicChannel.Id}),
+		)
+		require.Nil(t, th.App.Srv.Store.Token().Save(token))
+		_, err := th.App.AddUserToTeamByToken(rguest.Id, token.Token)
+		require.NotNil(t, err)
+		errRegexp := regexp.MustCompile(`Email must be from a specific domain`)
```

Comment:
The error is an AppError. Let's check the id rather than do a regex match which is a bit brittle IMO. And also change the check to assert.Equal from require.Equal.

Reply:
Oh, didn't know about the err.Id. Much better indeed